posthog/clickhouse/client/execute_async.py

```diff
@@ -85,50 +86,58 @@ def execute_process_query(
 
     from posthog.api.services.query import process_query, ExecutionMode
     from posthog.models import Team
+    from posthog.models.user import User
 
     team = Team.objects.get(pk=team_id)
+    user = User.objects.get(pk=user_id)
 
-    query_status = manager.get_query_status()
+    with push_scope() as scope:
+        scope.set_user({"email": user.email, "id": user_id, "username": user.email})
+        scope.set_tag("team_id", team_id)
```

Comment:
We could move this `with` in the function that calls `execute_process_query`, to reduce concerns here and not have the nesting level. But I can also clean that up later at some point.